robot-server/robot_server/authentication/_dependencies.py

```
@@ -0,0 +1,76 @@
+from aiohttp.client import ClientSession
+from aiohttp.client_exceptions import ClientConnectionError
+from fastapi import Header, Depends, status
+from typing import Union
+from typing_extensions import Literal
+
+from robot_server.errors import ErrorDetails
+from server_utils.util import call_once
+from robot_server.versioning import get_requested_version
+
+_AUTH_TOKEN_MIN_VERSION = 5
+
+_SYSTEM_SERVER_BASE_URL = "http://localhost:32950"
+
+
+class AuthenticationFailed(ErrorDetails):
+    """An error if authentication token appears invalid."""
+
+    id: Literal["AuthenticationFailed"] = "AuthenticationFailed"
+    title: str = "Authentication Bearer authentication failed."
+
+
+class SystemServerConnectionError(ErrorDetails):
+    """An error if the system server cannot be found, and is required."""
+
+    id: Literal["SystemServerConnectionError"] = "SystemServerConnectionError"
+    title: str = "System Server connection error, cannot authenticate."
+
+
+@call_once
+async def _get_system_server_client() -> ClientSession:
+    """Get a singleton client connection to the system server."""
+    try:
+        # TODO: The version of aiohttp we are using lacks type annotations,
+        # so we have to ignore mypy complaining about the initializer of this
+        # class.
+        return ClientSession()  # type: ignore
+    except Exception as e:
+        raise SystemServerConnectionError(detail=str(e)).as_error(
+            status.HTTP_403_FORBIDDEN
+        )
+
+
+async def check_auth_token_header(
+    authenticationBearer: Union[str, None] = Header(
+        None, description="Authentication token for the client."
+    ),
+    version: int = Depends(get_requested_version),
+) -> None:
+    """Get the request's auth token header and verify authenticity."""
+    if version < _AUTH_TOKEN_MIN_VERSION:
+        return
```

do we want to explicitly raise an error here with an unsupported API version or similar?

---

I think we specifically want to have backwards compatibility, at least to start. The `MIN_API_VERSION` in versioning.py handles the cutoff of errors for old API versions.